packages/integration/src/github/DefaultGithubCredentialsProvider.ts

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2021 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { GithubCredentials, GithubCredentialsProvider } from './types';
+import { ScmIntegrations } from '../ScmIntegrations';
+import { SingleInstanceGithubCredentialsProvider } from './SingleInstanceGithubCredentialsProvider';
+
+type SingleInstanceGithubCredentialsProviderCollection = {
```

Comment:
Bit of a nit but this name doesn't quite match the type. Tbh I think an inline `Map<string, GithubCredentialsProvider>` would be the way to go here

Reply:
Good idea.